ios/Classes/Plugin/PluginController.swift

```diff
@@ -221,6 +221,12 @@ final class PluginController {
                 print("Warning! No event channel set up to report a connection failure: \(error)")
                 return
             }
+            
+            sink.add(.success(DeviceInfo.with {
+                $0.id = args.deviceID
+                $0.connectionState = encode(.connecting)
+            }
+            ))
```

Comment:
This is not the right place. Here we know an error has already happened while we were trying to connect.

`.connecting` should before or right after the call to `central.connect` (depends on whether we want the errors to be reported after `.connecting` or without it).

Are there other similar cases around (maybe scanning)? Do you know how it's happening on Android?

---

A separate issue is formatting, it looks off, at least in the end (the closing braces). Look how it's formatted somewhere else, please.

Reply:
@werediver  I agree with that the place is quite late. My rationale of putting it there was that  from here we have a sink that is guaranteed to be there (see the guard on the line above). However I can move that part of the code further up. What do you think? 

Other question is there a standard formatter in XCode that you use?

Reply:
Thanks for the help. I think I put it at the correct place now